cpp/include/cudf/utilities/traits.hpp

```diff
@@ -439,6 +472,20 @@ constexpr inline bool is_fixed_width(data_type type)
   return cudf::type_dispatcher(type, is_fixed_width_impl{});
 }
 
+/**
+ * @brief Indicates whether the type `T` is a "primitive type". Currently this
```

Comment:
What is a primitive type? I would have thought int, float, bool, etc. But this implies that other fixed-width types like durations and timestamps are primitive types?

Reply:
This is the name that @jrhemstad originally recommended - and I thought it was pretty good. Not sure what a better name would be. We had a `is_simple` previously that wasn't being used. But as you said - it is basically `fixed_width` not including `fixed_point`.

Reply:
Seems weird that timestamps and durations are primitive types. @jrhemstad had you thought about that?

Reply:
I did not intend to include timestamp/duration in `is_primtive`.

Reply:
Hmmm, my bad - the original location where you made the comment was just `fixed_width` minus `fixed_point`. If that is the case then maybe it is worth removing if for now until we have a use for it? Otherwise it won't be used anywhere in this PR.

Reply:
Removed.